### PR TITLE
Merge meta.yaml and devfile.yaml

### PR DESCRIPTION
**Pull request by GeekArthur**
Signed-off-by: jingfu wang <jingfu.j.wang@ibm.com>

### What does this PR do?
This PR adds new metadata fields to the existing devfile schema so that we can merge meta.yaml and devfile.yaml

### What issues does this PR fix or reference?
Fixes https://github.com/devfile/api/issues/172

### CODE REVIEW

**Review by johnmcollier — APPROVED** [review]
Schema changes look fine to me.

But... I think we also need some folks from the devworkspace team to take a look at this and make sure it's not gonna affect anything on their end

**Comment by GeekArthur** [comment]
> Schema changes look fine to me.
> 
> But... I think we also need some folks from the devworkspace team to take a look at this and make sure it's not gonna affect anything on their end

Agree, let me ask some folks to review it.
@davidfestal @amisevsk @sleshchenko Can you please help review this PR as I find you are the owner of this repo? Thanks.

File: pkg/devfile/header.go [hunk 1]
```diff
@@ -30,4 +30,24 @@ type DevfileMetadata struct {
...
+	// +optional
+	Tags []string `json:"tags,omitempty"`
+
+	// Optional devfile icon
+	// +optional
+	Icon string `json:"icon,omitempty"`
+
+	// Optional devfile global memory limit
+	// +optional
+	GlobalMemoryLimit string `json:"globalMemoryLimit,omitempty"`
```
Comment (amisevsk) [inline pkg/devfile/header.go:52]: I'm not sure if `GlobalMemoryLimit` is needed (but I'm also not opposed to adding it). The story behind this field is that we needed a way to get the total memory required by a workspace, and this was hard to do from a very simple registry -- plugin memory requirements are stored in their own meta.yamls and can change arbitrarily while the devfile registry is running, and further the devfile itself can override the values set in the plugin. This field is *manually* set by the devfile author to smooth a use-case in Che, but the value here can drift apart from reality (if e.g. one of your plugins updates and starts using more memory). 

I wonder if the new devfile registry could offer some metrics/calculation to get this number directly without user input?

**Review by amisevsk — APPROVED** [review]
LGTM generally.

**Review by sleshchenko — APPROVED** [review]
Nothing to add apart from Angel's comment

File: pkg/devfile/header.go (hunk 1 above)
Comment (GeekArthur) [reply pkg/devfile/header.go:52]: Thanks for bringing that up, AFAIK currently the new devfile registry/OCI devfile registry doesn't provide a mechanism to directly get the number of global memory limit, so for the timing being, let's still add this field. Given this PR is only for purely migrating fields from meta.yaml, I think we can merge this PR for now, then open separate issue to discuss the existence of the specific field if it's needed as I think that may or may not bring more discussion and design.

**Pull request merged**